worlds/messenger/__init__.py

```diff
@@ -145,14 +146,15 @@ def generate_early(self) -> None:
 
         self.shop_prices, self.figurine_prices = shuffle_shop_prices(self)
 
-        starting_portals = ["Autumn Hills", "Howling Grotto", "Glacial Peak", "Riviere Turquoise", "Sunken Shrine", "Searing Crags"]
+        starting_portals = ["Autumn Hills", "Howling Grotto", "Glacial Peak", "Riviere Turquoise", "Sunken Shrine",
+                            "Searing Crags"]
         self.starting_portals = [f"{portal} Portal"
                                  for portal in starting_portals[:3] +
                                  self.random.sample(starting_portals[3:], k=self.options.available_portals - 3)]
 
         # super complicated method for adding searing crags to starting portals if it wasn't chosen
-        # TODO add a check for transition shuffle when that gets added back in
-        if not self.options.shuffle_portals and "Searing Crags Portal" not in self.starting_portals:
+        if (not self.options.shuffle_portals and not self.options.shuffle_transitions
+                and "Searing Crags Portal" not in self.starting_portals):
```

Comment:
Just documenting this here. There's a problem where it very often has an empty sphere 1 with only the three always-open portals and transition shuffle, removing this increases the generation success rate dramatically
```suggestion
        if not self.options.shuffle_portals and "Searing Crags Portal" not in self.starting_portals:
```